### DISCUSSION TITLE
add a href link to _category_.json in autogenerated mode

### DISCUSSION THREAD

**Discussion by pincman** [facebook/docusaurus#6836] [General]
how to do it?

**Comment by Josh-Cena** [comment]
This? https://docusaurus.io/docs/sidebar/autogenerated#autogenerated-sidebar-metadata

**Comment by pincman** [comment]
@Josh-Cena 
that's not it
currently my configuration is like this

```javascript
const sidebars = {
    manuals: [{ type: 'autogenerated', dirName: 'manuals' }],
};

module.exports = sidebars
```

and which in docs dir like this

```sh
└── manuals
    ├── frontend
    │   └── _category_.json
    ├── go
    │   ├── _category_.json
    │   ├── test1.md
    │   └── test3.md
    ├── index.md
    ├── node
    │   └── _category_.json
    └── php
        └── _category_.json
```

content of `go/_category_.json`

```json
{
    "label": "golang",
    "position": 4,
    "items": [
        {
            "type": "link",
            "label": "test",
            "href": "https://test.com",
             "position": 2
        }
    ]
}
```
but it not work,error log output by console
```sh
 ValidationError: "items" is not allowed
```

**Reply by Josh-Cena** [reply]
You cannot add additional items to an autogenerated slice (see #5689). For now, if you want to add links to any category, you have to make the entire category non-autogenerated, as a `type: 'category'` item consisting of autogenerated children. Like this:

```js
const sidebars = {
  manuals: [
    {
      type: "category",
      label: "golang",
      items: [
        { type: "autogenrated", dirName: "go" },
        { type: "link", label: "test", href: "https://test.com" },
      ],
    },
    // Go on to declare categories for node and php
  ],
};
```

**Reply by slorber** [reply]
technically it's possible to add extra content to autogenerated categories by writing your own generator: https://docusaurus.io/docs/sidebar/autogenerated#customize-the-sidebar-items-generator

but for sure it's not the most convenient API to do so

**Reply by pincman** [reply]
> ```js
> { type: "autogenrated", dirName: "go" }
> ```

can i exclude some file like  `{ type: "autogenrated", dirName: "go",exclude: ['index.md'] }` for it?

**Reply by Josh-Cena** [reply]
That would also be #5689. It's something we'd like to focus on right next, given the wide need to exclude the index page from the subitems, but it's currently not possible. You can also try @slorber's solution to enhance the generator and insert extra links where you want.

**Comment by mohab-sameh** [comment]
Also need this. Any idea if this is planned soon? It's much more convenient to use the auto-generated sidebar, but external links are also needed.

**Comment by hadar-co** [comment]
@Josh-Cena @slorber Hey guys, I also need this capability, any plans to add it? Thanks

**Comment by jarodtaylor** [comment]
Is this still not an option? I tried changing the link the _category_json and it's a no-go.

**Reply by slorber** [reply]
The issue to track is https://github.com/facebook/docusaurus/issues/5689

I'm locking this discussion because all related discussions should be added to the linked issue above. 

We still don't have a first-class option, but there are workarounds
